tests/jballerina-unit-test/src/test/resources/test-src/annotations/local_record_annotation_test.bal

```diff
@@ -0,0 +1,181 @@
+                                };
+    func();
+}
+
+function testAnnotationOnLocalRecordWithinLambdaFunction1() {
+    string k = "chiranS";
+    function(string) func = function (string value) {
+                                    record {@annot {value: value} string x;} r = {x : ""}; 
+                                    any m = getLocalRecordAnnotations(typeof r, "$field$.x");
+                                    assertEquality("typedesc map", (typeof m).toString());
+                                    assertEquality("{\"annot\":{\"value\":\"chiranS\"}}", m.toString()); 
+                                };
+    func(k);
+}
+
+function testAnnotationOnLocalRecordWithinNestedLambdaFunctions() {
+    string k = "chiranS";
+    function () returns function() var1 = function () returns function() {
+                                function() func = function () {
+                                    record {@annot {value: k} string x;} r = {x : ""}; 
+                                    any m = getLocalRecordAnnotations(typeof r, "$field$.x");
+                                    assertEquality("typedesc map", (typeof m).toString());
+                                    assertEquality("{\"annot\":{\"value\":\"chiranS\"}}", m.toString()); 
+                                };
+                                return func;
+                                };
+    function() func = var1();
+    func();
+}
+
+function testAnnotationOnLocalRecordWithinNestedLambdaFunctions1() {
+    string k = "chiranS";
+    function (string) returns function() var1 = function (string val) returns function() {
+                                function() func = function () {
+                                    record {@annot {value: val} string x;} r = {x : ""}; 
+                                    any m = getLocalRecordAnnotations(typeof r, "$field$.x");
+                                    assertEquality("typedesc map", (typeof m).toString());
+                                    assertEquality("{\"annot\":{\"value\":\"chiranS\"}}", m.toString()); 
+                                };
+                                return func;
+                                };
+    function() func = var1(k);
+    func();
+}
+
+function testAnnotationOnLocalRecordWithinNestedLambdaFunctions2() {
+    string k = "chiranS";
+    function (string) returns function() var1 = function (string val) returns function() {
+                                string name = "Name: ";
+                                function() func = function () {
+                                    record {@annot {value: name + val} string x;} r = {x : ""}; 
+                                    any m = getLocalRecordAnnotations(typeof r, "$field$.x");
+                                    assertEquality("typedesc map", (typeof m).toString());
+                                    assertEquality("{\"annot\":{\"value\":\"Name: chiranS\"}}", m.toString()); 
```

Comment:
Let's assert the value, and not the string representation. We lose type information when asserting as a string.